Enable cri-containerd-tests for arm64

seungukshin opened the pull request:
This change enables cri-containerd-tests for arm64. The edk2 is required for memory hot plug on qemu for arm64. So this makes sure to build edk2 and adds it to configuration-qemu.toml for arm64.

This adds one workflow to `ci.yaml` which already has 19 workflows. (GitHub limits 20 workflows) `run-cri-containerd-tests-arm64` will be merged with `run-cri-containerd-tests-s390x` and `run-cri-containerd-tests-ppc64le` in a following PR like [this PR](https://github.com/kata-containers/kata-containers/pull/10932).

RuoqingHe commented on tools/packaging/kata-deploy/local-build/kata-deploy-binaries.sh:
```diff
@@ -902,6 +902,10 @@ install_ovmf_sev() {
 	install_ovmf "sev" "edk2-sev.tar.gz"
 }
 
+install_ovmf_arm64() {
+	install_ovmf "arm64" "edk2-arm64.tar.gz"
+}
+
```
Can we detect architecture inside `install_ovmf` instead of calling it 🤔

RuoqingHe commented on tools/packaging/static-build/ovmf/build-ovmf.sh:
```diff
@@ -65,11 +68,15 @@ info "Done Building"
 
 build_path_target_toolchain="Build/${package_output_dir}/${build_target}_${toolchain}"
 build_path_fv="${build_path_target_toolchain}/FV"
-stat "${build_path_fv}/OVMF.fd"
 if [ "${ovmf_build}" == "tdx" ]; then
 	build_path_arch="${build_path_target_toolchain}/X64"
 	stat "${build_path_fv}/OVMF_CODE.fd"
 	stat "${build_path_fv}/OVMF_VARS.fd"
+elif [ "${ovmf_build}" == "arm64" ]; then
+	stat "${build_path_fv}/QEMU_EFI.fd"
+	stat "${build_path_fv}/QEMU_VARS.fd"
+else
+	stat "${build_path_fv}/OVMF.fd"
```
With this change `stat OVMF.fd` is skipped for `tdx` variant, is that expected 🧐

seungukshin replied:
I thought `tdx` only uses `OVMF_CODE.fd` and `OVMF_VARS.fd` and `snp` uses `OVMF.fd` because I refered to the old version. I confirmed `tdx` also uses `OVMF.fd` in the latest version, so I'll fix this.

RuoqingHe reviewed (commented):
Some suggestions 🙂

RuoqingHe commented:
Thank @seungukshin for this work - but this touches our CI, we need ACs to take a look here

kevinzs2048 commented:
Thanks @seungukshin, I want to submit a PR to build AAVF but looks that you already cover this, great!

kevinzs2048 commented on tools/packaging/static-build/ovmf/build-ovmf.sh:
```diff
@@ -65,11 +68,16 @@ info "Done Building"
+	stat "${build_path_fv}/OVMF.fd"
 	stat "${build_path_fv}/OVMF_CODE.fd"
 	stat "${build_path_fv}/OVMF_VARS.fd"
+elif [ "${ovmf_build}" == "arm64" ]; then
```
On Qemu Arm64 it expects the 64M  CODE and VARS files, it would be better to follow the debian build script for AAVMF(qemu-efi-arm64), so adding the truncate operation and rename to AAVMF would be great :-)
https://salsa.debian.org/qemu-team/edk2/-/blob/debian/latest/debian/rules?ref_type=heads#L223

seungukshin replied:
Sure, I'll apply renaming and truncating.

kevinzs2048 reviewed (commented):
Thanks for the support of build EDK2. Overall I like the idea about re-using the ovmf build script for AAVMF. As OVMF and AAVMF are almost shared the same build process, no needs to set up a separately build script for AAVMF as they just have slightly difference in file generation process.

stevenhorsman commented:
As this makes a workflow change we'll need to test this manually with a topic branch, so I'm putting `do-not-merge` on this. Once we've had some initial reviews and checks I can help create and test this based on this codebase.

stevenhorsman commented on .github/workflows/ci.yaml:
The patch format of this commit doesn't make the project guidelines so needs updating

seungukshin replied:
I'll add some description on this commit.

stevenhorsman commented on .github/workflows/run-cri-containerd-tests-arm64.yaml:
```diff
@@ -0,0 +1,53 @@
+      - name: get-kata-tarball
+        uses: actions/download-artifact@v4
+        with:
+          name: kata-static-tarball-s390x${{ inputs.tarball-suffix }}
```
I don't think this should be s390x

seungukshin replied:
I'll fix this.

stevenhorsman commented:
The test is failing on the topic branch (https://github.com/kata-containers/kata-containers/actions/runs/13855790737/job/38774441102):
```
go build  -gcflags=-trimpath=/home/github-runner/actions-runner/_work/kata-containers/kata-containers/src -buildmode=pie  -o bin/ctr -ldflags '-X github.com/containerd/containerd/v2/version.Version=v2.0.3.m -X github.com/containerd/containerd/v2/version.Revision=06b99ca80cdbfbc6cc8bd567021738c9af2b36ce.m -X github.com/containerd/containerd/v2/version.Package=github.com/containerd/containerd/v2 -s -w ' -tags "no_btrfs"  ./cmd/ctr
go: errors parsing go.mod:
/home/github-runner/actions-runner/_work/kata-containers/kata-containers/containerd/go.mod:3: invalid go version '1.22.0': must match format 1.23
make: *** [Makefile:264: bin/ctr] Error 1
[integration-tests.sh:161] ERROR: sudo -E PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/usr/local/sbin:/usr/local/go/bin:/usr/local/bin REPORT_DIR=/tmp/test-cri-containerd.xAf4 FOCUS=TestImageLoad RUNTIME= CONTAINERD_CONFIG_FILE=/tmp/test-cri-containerd.xAf4/test-containerd-config.temp make GO_BUILDTAGS=no_btrfs -e cri-integration
Error: Process completed with exit code 2.
```
I'm not sure I understand why the containerd tests are different for arm vs the other platforms?

kevinzs2048 commented on .github/workflows/run-cri-containerd-tests-arm64.yaml:
```diff
@@ -0,0 +1,53 @@
+      matrix:
+        containerd_version: ['active']
+        vmm: ['qemu']
+    runs-on: arm64-k8s
```
@stevenhorsman @fidencio Could we add another label here? I just found that the job is scheduling to the same node which is running k8s-test.
We have 2 machines, it's better to use jade01 as the k8s-test, and the other one jade02 can serve for the Containerd test.

stevenhorsman replied:
Is using the existing `arm64-non-k8s` label here from jade02 sufficient?

stevenhorsman replied:
BTW as a reference point the k8s and containerd tests run on the same labels for s390x, and can run on the same nodes for powerpc, so I'm not sure if there is some extra clean up that means the containerd test and k8s tests should be okay on the same node?

kevinzs2048 replied:
That's good, just make sure k8s-test to jade01 and contained job come to jade02 is fine.
Thanks

kevinzs2048 replied:
@stevenhorsman Aha, I'm not aware of that. If so the job can be running on either node, so let me fix the containerd build issue first in the container job first.

stevenhorsman replied:
Just to be clear, it is not mandatory to run them both on the same node, so I don't have an issue with this being run on jade02 only, I just wanted to share that information in case there is something in a pre/post-install script that we might want to look into

kevinzs2048 replied:
@stevenhorsman Thanks, I understand. I update the golang version to 1.23 now, please help to re-trigger the test. I test locally it works for build now.

seungukshin replied:
@kevinzs2048 Do you know why go1.23 is required for arm64 only? because amd64 does not have any issue with go1.22.11 - https://github.com/kata-containers/kata-containers/actions/runs/13855790737/job/38774002963#step:4:169

kevinzs2048 replied:
@stevenhorsman That would be great, we can separate the tasks in the short term.

stevenhorsman replied:
On re-run we get:
https://github.com/kata-containers/kata-containers/actions/runs/13903378878/job/38903891506
```
Warning: Unable to clean or reset the repository. The repository will be recreated instead.
Deleting the contents of '/home/github-runner/actions-runner/_work/kata-containers/kata-containers'
Error: File was unable to be removed Error: EACCES: permission denied, unlink '/home/github-runner/actions-runner/_work/kata-containers/kata-containers/containerd/bin/containerd'
```
so I guess that jade-02 needs a pre-action script that removes the workspace files?

seungukshin replied:
@stevenhorsman , thank you for testing!
@kevinzs2048 , do you have any clean up script for `arm64-non-k8s` or should it be enough to remove `kata-containers` directory for clean up?

kevinzs2048 replied:
I've removed the kata-containers directory now. Not sure if this is manually download for test or not. Please help to re-triagger. Thanks!

seungukshin replied:
I thought it would be executed as root user, but it was not the case. I'll add `sudo` to `apt-get` and `rm` commands.

seungukshin replied:
Hi @stevenhorsman , sorry for bothering you frequently, but could you retrigger [this test](https://github.com/kata-containers/kata-containers/actions/runs/13903378878/job/38947202674)?

seungukshin replied:
Hi @stevenhorsman ,
I added cleaning up step right before starting tests as following:
```sh
        info "Clean up containers and pods"
	restart_containerd_service
	containers=( $(sudo crictl ps --all -o json | jq -r '.containers[].id') )
	for c in "${containers[@]}"; do
		sudo crictl rm -f $c
	done
	pods=( $(sudo crictl pods -o json | jq -r '.items[].id') )
	for p in "${pods[@]}"; do
		sudo crictl rmp -f $p
	done
```
Could you retrigger the test?

seungukshin replied:
@stevenhorsman , Sorry for bothering you several times. The new clean-up logic is required if there are remained containers or pods from the previous failed test in a bare metal. (the test cleans up all containers or pods in success, but it leaves some in failure) We can move this clean-up logic to end of the test.

@kevinzs2048 , It seems the GitHub runner does not remove work directory. I found the following script [here](https://github.com/orgs/community/discussions/51329). Could you confirm if `runner_base_dir` is `/home/github-runner/actions-runner/_work` and test it if possible?
```sh
#!/bin/bash

set -e

if [ -z "${GITHUB_WORKSPACE}" ]; then
  echo "ERROR: GITHUB_WORKSPACE is not set"
  exit 1
fi

workspace_dir=$(realpath "$GITHUB_WORKSPACE")
runner_base_dir=/home/github-runner/actions-runner/_work

if [[ $workspace_dir == $runner_base_dir* ]]; then
  find $GITHUB_WORKSPACE -mindepth 1 -delete
else
  echo "ERROR: GITHUB_WORKSPACE ($GITHUB_WORKSPACE) must be within $runner_base_dir"
  exit 1
fi
```

kevinzs2048 replied:
For  https://github.com/kata-containers/kata-containers/actions/runs/13944024353/job/39029907765, We hit a corner case: https://github.com/containerd/containerd/issues/7247

kevinzs2048 commented:
Hi @seungukshin @stevenhorsman, regarding the issue for `invalid go version '1.22.0': must match format 1.23`, here the version `1.23` is not specified version, it is just a `format` without the last `.0`

I think this comment can be a precise description: https://github.com/siderolabs/conform/issues/251#issuecomment-2523243166
```
First, check the Go version you have installed with go version.

If you are using Go 1.20 or older, your project's go.mod has to specify the version without the patch number, so for example go 1.19.

If you are using Go 1.21 or newer, your project's go.mod can specify the version with or without the patch number, both seem to work.
```

In the Arm64 node, previously we used Golang 1.18 before, which is the same condition that asks for a version name as `1.23`. Now both the 2 nodes are re-installed the golang as 1.22.11 already.

seungukshin commented:
Hi @kevinzs2048 ,

Go version 1.22.11 is installed as a part of cri-containerd test. (https://github.com/kata-containers/kata-containers/actions/runs/13855790737/job/38774002963#step:4:169) The version is defined in [here](https://github.com/kata-containers/kata-containers/blob/f308cbba93acafca5baee12d11466a5ab09508e8/versions.yaml#L385).

The test on amd64 with go version 1.22.11 passed (https://github.com/kata-containers/kata-containers/actions/runs/13855790737/job/38774002963) and my manual test on arm64 (ubuntu-22.04 on ampere) with go version 1.22.11 also passed.

I found the same error message in local test when I use go version 1.18. Is there any chance that go version 1.18 is remained and it is used for building containerd in `arm64-k8s` environment?

kevinzs2048 commented:
> Hi @kevinzs2048 ,
> 
> Go version 1.22.11 is installed as a part of cri-containerd test. (https://github.com/kata-containers/kata-containers/actions/runs/13855790737/job/38774002963#step:4:169) The version is defined in [here](https://github.com/kata-containers/kata-containers/blob/f308cbba93acafca5baee12d11466a5ab09508e8/versions.yaml#L385).
> 
> The test on amd64 with go version 1.22.11 passed (https://github.com/kata-containers/kata-containers/actions/runs/13855790737/job/38774002963) and my manual test on arm64 (ubuntu-22.04 on ampere) with go version 1.22.11 also passed.
> 
> I found the same error message in local test when I use go version 1.18. Is there any chance that go version 1.18 is remained and it is used for building containerd in `arm64-k8s` environment?

Hi @seungukshin , I don't think Go 1.18 is specially needed, they are probably the legacy remaining there.
The reason why I advise we separate the containerd and K8s job is because that the containerd job destroying the pre-installed K8s and containerd every time, and we have not adding the pre-setup/clean jobs for changing to a new job.

seungukshin commented:
Hi @kevinzs2048 ,

Thank you for taking a look into this problem.

> Btw not sure if this is a common conditions on containerd 2.0.x, just look at the containerd log with 2.0.x (CI job for cri-containerd, active, qemu)in other architecture, I also did not find any enablement status on kata-qemu either: https://github.com/kata-containers/kata-containers/actions/runs/13982959903/job/39154518870, the containerd log shows:

You're right. It seems the default runtime was not changed according to the log. But, what I cannot understand is it should fail if the runc is used because the runc container shows host memory size instead of 2GB, but the test on amd64 passed.

To confirm this, I added [this commit](https://github.com/seungukshin/kata-containers/commit/0d773148af87332164d783793a3a58a25260780a) and triggered the test.
```diff
modified   tests/integration/cri-containerd/integration-tests.sh
@@ -200,6 +200,7 @@ EOF
         sudo cp $CONTAINERD_CONFIG_FILE "$default_containerd_config"

         restart_containerd_service
+        systemctl status containerd

         sudo crictl pull $image
         podid=$(sudo crictl runp $pod_yaml)
```

The test installed [the containerd (v2.0.4)](https://github.com/seungukshin/kata-containers/actions/runs/14035218869/job/39294353683#step:4:180) on [`/usr/local/bin/containerd`](https://github.com/kata-containers/kata-containers/blob/8f1e2f7b1e3c2fa989d08b7284993bd030e2fdcc/tests/common.bash#L679), but [`/usr/bin/containerd`](https://github.com/seungukshin/kata-containers/actions/runs/14035218869/job/39294353683#step:7:1064) was used for the memory update test. It probably came from the ubuntu-22.04's package and its version would be [1.7.24](https://launchpad.net/ubuntu/jammy/+package/containerd) which could unerstand `[plugins.cri.containerd]` (so, memory update test passed)

So, I'll
1. add a step to remove the existing containerd version to test the correct containerd version.
2. apply kevin's change for containerd v2.0.x can understand.
3. keep steps to clean up the existing go version and pods just in case.

seungukshin commented:
* added a step to remove preivous GitHub workspace ([change](https://github.com/kata-containers/kata-containers/pull/11019/files#diff-a0ab5d165ca6565e4cbbc2531f06809e2ecf0760d4cf2f7e36657df520c0cfb0R29))
* added a step to remove existing containerd to test the correct containerd version ([change](https://github.com/kata-containers/kata-containers/pull/11019/files#diff-1f5ed05ffe809660ba45565db7641a0973fd4daac2d62971f3e9e7ae786d7ef0R26))
* changed the containerd config file from `cri` to `io.containerd.grpc.v1.cri` for both v1.7.x and v2.0.x ([change](https://github.com/kata-containers/kata-containers/pull/11019/files#diff-69fe3659cef9894cbc7b5bc9e1fa60209b1a4c3be66d002a2571237a17b3e179R111))

kevinzs2048 commented:
@seungukshin Thanks for the quick response.
I've tested the configuration locally, it is working quite well. Appreciated!

Apart from this, have you met the error about with containerd 2.0.x
```
./script/test/utils.sh: line 172: CGROUP_DRIVER: unbound variable
```
And this has been fixed by: https://github.com/kata-containers/kata-containers/commit/644af52968bb588bcd9831632abbf5bc452f32c6, so it's better to rebase the PR.

seungukshin commented:
After rebase to the main, the following error happens with containerd v1.7.27 on both amd64 and arm64:
```
containerd: failed to load TOML from /tmp/test-cri-containerd.U3B9/test-containerd-config.temp: invalid plugin key URI "linux" expect io.containerd.x.vx
```
It seems it cannot recognize `[plugins.linux]` anymore, so it is changed to `[plugins."io.containerd.grpc.v1.linux"]`.

fidencio commented on tests/integration/cri-containerd/integration-tests.sh:
```diff
@@ -104,24 +104,27 @@ function create_containerd_config() {
 	info "Kata Config Path ${runtime_config_path}, Runtime Binary Name ${runtime_binary_path}"
 
 cat << EOF | sudo tee "${CONTAINERD_CONFIG_FILE}"
+version = 2
```
So, this part here is quite tricky, and what I'd do is the following.
* Run `containerd config default` and get the version from there
* Check whether the configuration is version 2 or version 3
* Adjust the fields accordingly to version 2 or version 3
  * You can use this for inspiration: https://github.com/kata-containers/kata-containers/blob/54dcf0d34293b2faea6593d7062fa3c286c26a86/tools/packaging/kata-deploy/scripts/kata-deploy.sh#L596-L618

seungukshin replied:
@fidencio , thank you for reviewing and testing this!
I used version 2 config because it can be used by both v1.7.x and v2.0.x, but it seems it might break some other tests.
I'll update the test config according to your suggestion.

fidencio replied:
Thanks! <3

seungukshin replied:
I updated as following:
```sh
        # check containerd config version
        if containerd config default | grep -q "version = 3\>"; then
                pluginid=\"io.containerd.cri.v1.runtime\"
        else
                pluginid="cri"
        fi
        info "Kata Config Path ${runtime_config_path}, Runtime Binary Name ${runtime_binary_path}"

cat << EOF | sudo tee "${CONTAINERD_CONFIG_FILE}"
[debug]
  level = "debug"
[plugins]
  [plugins.${pluginid}]
    [plugins.${pluginid}.containerd]
        default_runtime_name = "$runtime"
      [plugins.${pluginid}.containerd.runtimes.${runtime}]
        runtime_type = "${runtime_type}"
        sandboxer = "${SANDBOXER}"
        $( [ $kata_annotations -eq 1 ] && \
        echo 'pod_annotations = ["io.katacontainers.*"]' && \
        echo '        container_annotations = ["io.katacontainers.*"]'
        )
        [plugins.${pluginid}.containerd.runtimes.${runtime}.options]
          ConfigPath = "${runtime_config_path}"
          BinaryName = "${runtime_binary_path}"
$( [ -n "$containerd_shim_path" ] && \
echo "[plugins.linux]" && \
echo "  shim = \"${containerd_shim_path}\""
)
EOF
```

After removing `version = 2`, the containerd does not complain about `plugins.linux`. So it is reverted.

fidencio reviewed (changes requested):
I've left a comment to ensure we don't break tests with containerd using config version = 3

seungukshin commented:
It still fails on `run-containerd-sandboxapi` test cases with following message ([link](https://github.com/kata-containers/kata-containers/actions/runs/14103999977/job/39508526042)):
```
E0327 10:52:07.442342   20383 remote_runtime.go:132] RunPodSandbox from runtime service failed: rpc error: code = Unknown desc = failed to create sandbox "83773ec1d7a374395cd2fef88610db915c22368d8b18bbcc71e7f103b2bc4efe": failed to start new shim for sandbox 83773ec1d7a374395cd2fef88610db915c22368d8b18bbcc71e7f103b2bc4efe: start failed: io.containerd.runc.v2: open config.json: no such file or directory: exit status 1
    main_test.go:266: 
        	Error Trace:	/home/runner/work/kata-containers/kata-containers/containerd/integration/main_test.go:266
        	            				/home/runner/work/kata-containers/kata-containers/containerd/integration/image_load_test.go:77
        	Error:      	Received unexpected error:
        	            	rpc error: code = Unknown desc = failed to create sandbox "83773ec1d7a374395cd2fef88610db915c22368d8b18bbcc71e7f103b2bc4efe": failed to start new shim for sandbox 83773ec1d7a374395cd2fef88610db915c22368d8b18bbcc71e7f103b2bc4efe: start failed: io.containerd.runc.v2: open config.json: no such file or directory: exit status 1
```

It seems it is not tested correctly until now because `snadboxer` setting is ignored because incorrect key (`[plugins.cri.containerd]`) with containerd v2.0.x. ([log from other topic branch](https://github.com/kata-containers/kata-containers/actions/runs/13815236394/job/38649524713#step:7:590) said not `shim` but `podsandbox` is still used for `sandboxer`)

I'm still looking into this problem.

seungukshin commented:
It seems containerd's integration tests fail with `sandboxer=shim` because `Spec` is not generated correctly. (I created an issue about this on containerd repo. - https://github.com/containerd/containerd/issues/11640)

I'll disable `run-containerd-sandboxapi` test cases refering above issue for this PR, and I'll re-enable them once above issue is fixed.

seungukshin commented:
Updated the PR with
* `run-containerd-sandboxapi` test cases are disabled with `TODO` tag refering https://github.com/containerd/containerd/issues/11640
* Removing both `containerd` and `containerd.io` packages not to use old version for testing becuase `ubuntu-22.04` uses `containerd.io`

Tested on the forked repo. for `basic-amd64-tests` (https://github.com/seungukshin/kata-containers/actions/runs/14264649776)
* `build-and-publish-tee-confidential-unencrypted-image` and `publish-csi-driver-amd64` failed because of ghcr permission
* `run-k8s-tests` failed because of azure permission

@fidencio , could you test this on the topic branch when you're availabe?

fidencio commented:
I missed the notification, please, get it updated (no conflicts) and I will test it later Today.

fidencio commented:
tests running on topic branch: https://github.com/kata-containers/kata-containers/actions/runs/15067550533

seungukshin commented:
run-cri-containerd-tests on ppc64le failed. I'm looking into it (https://github.com/kata-containers/kata-containers/actions/runs/15067550533/job/42356974496)

fidencio commented:
That's most likely not your fault.  I failed to find the arm64 ones, though. Can you link them here as well?

seungukshin commented:
run-cri-containerd-tests on s390x also failed - https://github.com/kata-containers/kata-containers/actions/runs/15067550533/job/42360225504
I'm still looking for run-cri-containerd-tests on arm64
(create-kata-tarball for arm64 - https://github.com/kata-containers/kata-containers/actions/runs/15067550533/job/42357540673)

stevenhorsman commented on .github/workflows/run-cri-containerd-tests.yaml:
```diff
@@ -0,0 +1,84 @@
+      GOPATH: ${{ github.workspace }}
+      KATA_HYPERVISOR: ${{ inputs.vmm }}
+    steps:
+      - name: Clean up workspace
+        run: |
+          if [ -z "${GITHUB_WORKSPACE}" ]; then
+            echo "ERROR: GITHUB_WORKSPACE is not set"
+            exit 1
+          fi
+
+          workspace_dir=$(sudo realpath "$GITHUB_WORKSPACE")
+          runner_base_dir=/home/github-runner/actions-runner/_work
+
+          if [[ $workspace_dir == $runner_base_dir* ]]; then
+            sudo find $GITHUB_WORKSPACE -mindepth 1 -delete
+          else
+            echo "ERROR: GITHUB_WORKSPACE ($GITHUB_WORKSPACE) must be within $runner_base_dir"
+            exit 1
+          fi
```
Shouldn't this be in the ACTIONS_RUNNER_HOOK_JOB_STARTED hook?

seungukshin replied:
I think it's reasonable to add this as a pre-job script.
@kevinzs2048 , could you add this script as a pre-job script on arm64-non-k8s?

kevinzs2048 replied:
@seungukshin Done, please help to check.

seungukshin replied:
Thanks @kevinzs2048 , I'll update this PR soon.

stevenhorsman commented on .github/workflows/basic-ci-amd64.yaml:
```diff
@@ -65,6 +65,8 @@ jobs:
       matrix:
         containerd_version: ['active']
         vmm: ['dragonball', 'cloud-hypervisor', 'qemu-runtime-rs']
+    # TODO: enable me when https://github.com/containerd/containerd/issues/11640 is fixed
+    if: false
```
Why do these jobs just need skipping now - is it related to the changes made in integration-tests exposing a bug?

seungukshin replied:
https://github.com/kata-containers/kata-containers/pull/11019#issuecomment-2758617135
You're correct. Until now `sandboxer=shim` was not tested with containerd v2.0.x because `[plugins.cri.containerd]` was ignored by containerd v2.0.x. This PR changes `[plugins.cri.containerd]` to `[plugins."io.containerd.cri.v1.runtime".containerd]` for containerd v2.0.x, so containerd v2.0.x recognizes it. But the test fails even without kata-containers. (https://github.com/containerd/containerd/issues/11640)

seungukshin replied:
Sure, I'll create a separate commit for this.

stevenhorsman commented on .github/workflows/run-cri-containerd-tests.yaml:
Can we use this workflow for the amd64 and s390x jobs as well?

seungukshin replied:
Sure, I'll make them use `run-cri-containerd-tests.yaml` and remove `run-cri-containerd` from `basic-ci-amd64.yaml` and `basic-ci-s390x.yaml`.

stevenhorsman replied:
I think you could call `run-cri-containerd-tests.yaml` from in `basic-ci-amd64.yaml` and `basic-ci-s390x.yaml` in case that's not what you are thinking, and saving some duplication would be great. Thank you!

stevenhorsman replied:
Yeah, that's equally valid, I just figured that keeping the different arch tests together might help if we want to split out testing of individual architectures in future

seungukshin replied:
It seems to use `run-cri-containerd-tests.yaml` from yaml file other than `ci.yaml` counts separately. (https://github.com/seungukshin/kata-containers/actions/runs/15109784032, counted as 23 workflows)

* failed because of 23 workflows
  * use `run-cri-containerd-tests.yaml` from `ci.yaml` 2 times (ppc64le and arm64)
  * use `run-cri-containerd-tests.yaml` from `basic-ci-amd64.yaml` (amd64)
  * use `run-cri-containerd-tests.yaml` from `basic-ci-s390x.yaml` (s390x)
* passed
  * use `run-cri-containerd-tests.yaml` from `ci.yaml` 4 times (amd64, s390x, ppc64le and arm64)

I'll use `run-cri-containerd-tests.yaml` from only `ci.yaml` and remove `run-cri-containerd-tests` jobs from `basic-ci-amd64.yaml` and `basic-ci-s390x.yaml`.

stevenhorsman replied:
That seems strange to me that the same calls are counted differently, but maybe it count all unique instances of the call from within each workflow as a single one?

seungukshin commented:
Updated the PR with

* rebased.
* split the commit for disabling `run-containerd-sandboxapi`.
* move cleaning up the workspace to `ACTIONS_RUNNER_HOOK_JOB_STARTED`.

The branch is tested on the forked repository (https://github.com/seungukshin/kata-containers/actions/runs/15110424692), but only for x86 because the forked repository does not have self-hosted runner for s390x, ppc64le and arm64.

@fidencio , could you test this on the topic branch when you're availabe?

seungukshin commented:
Updated the PR with

* rebased.
* add a job name for run-cri-containerd - `run-cri-containerd-<arch> (<containerd_version>, <vmm>)`

The branch is tested on the forked repository (https://github.com/seungukshin/kata-containers/actions/runs/15325361051/job/43128493358), but only for x86 because the forked repository does not have self-hosted runner for s390x, ppc64le and arm64.

@fidencio , could you test this on the topic branch when you're availabe?

seungukshin commented:
Updated the PR with

* rebased.
* add conditions on `run-cri-containerd-amd64` and `run-cri-containerd-s390x`

The branch is tested on the forked repository (https://github.com/seungukshin/kata-containers/actions/runs/15419894669/job/43395926563), but only for x86 because the forked repository does not have self-hosted runner for s390x, ppc64le and arm64.

@fidencio , could you test this on the topic branch when you're availabe?

stevenhorsman commented:
It looks like the arm64 containerd job is working there: https://github.com/kata-containers/kata-containers/actions/runs/15559004799/job/43809404576

seungukshin commented:
run-containerd-tests on s390x failed. I'm looking into it: https://github.com/kata-containers/kata-containers/actions/runs/15559004799/job/43814957726

BbolroC commented:
@stevenhorsman Please mark the failing test as not-required as reported at https://katacontainers.slack.com/archives/C01HUGDR0CE/p1749558577846129. Thanks!

seungukshin commented on .github/workflows/ci.yaml:
```diff
@@ -416,11 +416,86 @@ jobs:
+    strategy:
+      matrix:
+        params: [
+          { runner: s390x-large,   arch: s390x,   containerd_version: active, vmm: qemu             },
+          { runner: s390x-large,   arch: s390x,   containerd_version: active, vmm: qemu-runtime-rs  },
```
I'll remove `runner` from the `inputs` and figure it out from the `arch` in `cri-containerd-tests.yaml`.

seungukshin commented on tests/integration/cri-containerd/integration-tests.sh:
```diff
@@ -101,27 +101,36 @@ function create_containerd_config() {
           BinaryName = "${runtime_binary_path}"
-[plugins.linux]
-       shim = "${containerd_shim_path}"
+$( [ -n "$containerd_shim_path" ] && \
```
I'll fix it

stevenhorsman commented on .github/workflows/ci.yaml:
```diff
@@ -416,11 +416,86 @@ jobs:
+    strategy:
+      matrix:
+        params: [
+          { runner: ubuntu-22.04,  arch: amd64,   containerd_version: lts,    vmm: clh              },
+          { runner: ubuntu-22.04,  arch: amd64,   containerd_version: lts,    vmm: dragonball       },
+          { runner: ubuntu-22.04,  arch: amd64,   containerd_version: lts,    vmm: qemu             },
+          { runner: ubuntu-22.04,  arch: amd64,   containerd_version: lts,    vmm: stratovirt       },
+          { runner: ubuntu-22.04,  arch: amd64,   containerd_version: lts,    vmm: cloud-hypervisor },
+          { runner: ubuntu-22.04,  arch: amd64,   containerd_version: lts,    vmm: qemu-runtime-rs  },
+          { runner: ubuntu-22.04,  arch: amd64,   containerd_version: active, vmm: clh              },
+          { runner: ubuntu-22.04,  arch: amd64,   containerd_version: active, vmm: dragonball       },
+          { runner: ubuntu-22.04,  arch: amd64,   containerd_version: active, vmm: qemu             },
+          { runner: ubuntu-22.04,  arch: amd64,   containerd_version: active, vmm: stratovirt       },
+          { runner: ubuntu-22.04,  arch: amd64,   containerd_version: active, vmm: cloud-hypervisor },
+          { runner: ubuntu-22.04,  arch: amd64,   containerd_version: active, vmm: qemu-runtime-rs  },
```
I think having the runner in inputs is fine, but it can be hard-coded in this job

stevenhorsman commented on .github/workflows/ci.yaml:
```diff
@@ -413,11 +413,86 @@ jobs:
+    strategy:
+      matrix:
+        params: [
+          { containerd_version: active, vmm: qemu             },
```
Yep - agreed

lifupan commented:
Hi  @seungukshin 

Why do you skipped the sandboxapi test?

stevenhorsman commented:
> Hi @seungukshin
> 
> Why do you skipped the sandboxapi test?

Hi @lifupan - asked the same and so @seungukshin added an explanation to the commit message in https://github.com/kata-containers/kata-containers/pull/11019/commits/203e3af94b9fbf31f0130da2bf94a9a98e1dc0e2

seungukshin commented:
> > Hi @seungukshin
> > Why do you skipped the sandboxapi test?
> 
> Hi @lifupan - asked the same and so @seungukshin added an explanation to the commit message in [203e3af](https://github.com/kata-containers/kata-containers/commit/203e3af94b9fbf31f0130da2bf94a9a98e1dc0e2)

Thanks @stevenhorsman 

Hi @lifupan , sandboxapi test fails with `shim` sandboxer on containerd (without kata-containers), so I also opened [an issue](https://github.com/containerd/containerd/issues/11640) on containerd repo.
